src/helm/proxy/clients/critique_client.py

```diff
@@ -209,3 +212,155 @@ def make_critique_request(self, request: CritiqueRequest) -> CritiqueRequestResu
+            request = Request(
+                model=self._model_name,
+                prompt=prompt,
+                max_tokens=max_tokens,
+                echo_prompt=False,
+            )
+            requests.append(request)
+        return requests
+
+    def _execute_requests(self, requests: List[Request], num_respondents: int) -> List[List[RequestResult]]:
+        """Execute a list of requests and return the responses."""
+        responses: List[List[RequestResult]] = []
+        for request in requests:
+            responses.append([])
+            for i in range(num_respondents):
+                request_with_random = dataclasses.replace(request, random=str(i))
+                response: RequestResult = self._client.make_request(request_with_random)
+                responses[-1].append(response)
+        return responses
+
+    def _parse_completion_to_question_choice(self, completion: str) -> List[str]:
+        """Convert a completion to a list of answer represented by a list of capital letters."""
+        completion_parsed = completion.replace(" ", "").replace("\n", "").replace(".", "").upper()
+        answers = completion_parsed.split(",")
+        assert len(answers) >= 1, f"Invalid answer: {completion}. There are no answers once parsed: {answers}."
+        assert all(
+            [answer in string.ascii_uppercase for answer in answers]
+        ), f"Invalid answer: {completion}. Some answers are not capital letters, once parsed: {answers}."
+        return answers
+
+    def _multiple_choice_completion_to_answer(self, question: CritiqueQuestionTemplate, completion: Sequence) -> str:
+        """Convert a multiple choice completion to an answer."""
+        assert question.question_type == "multiple_choice"
+        answers: List[str] = self._parse_completion_to_question_choice(completion.text)
+        assert len(answers) == 1, f"Invalid answer: {completion}. Multiple choice questions should have one answer."
+        return answers[0]
+
+    def _checkbox_completion_to_answer(self, question: CritiqueQuestionTemplate, completion: Sequence) -> List[str]:
+        """Convert a checkbox completion to an answer."""
+        assert question.question_type == "checkbox"
+        answers: List[str] = self._parse_completion_to_question_choice(completion.text)
+        assert len(answers) <= len(
+            question.options
+        ), f"Invalid answer: {completion}. Checkbox questions should have at most one answer per option."
+        return answers
+
+    def _free_response_completion_to_answer(self, question: CritiqueQuestionTemplate, completion: Sequence) -> str:
+        """Convert a free response completion to an answer."""
+        assert question.question_type == "free_response"
+        return completion.text
+
+    def _letter_answer_to_actual_answer(
+        self, letter_answer: Union[str, List[str]], question: CritiqueQuestionTemplate, fields: Dict[str, str]
```

Comment:
Avoid `Union` types if possible, because they make it harder to reason about the code and introduce more cases that need to be handled

Reply:
I think this is needed here right? Otherwise, this will be handled at the call site which I think is worse.